common/src/java/org/apache/hive/common/util/TimestampParser.java

```diff
@@ -199,6 +205,19 @@ public Timestamp parseTimestamp(final String text) {
 
   }
 
+  public TimestampTZ parseTimestamp(String text, ZoneId defaultTimeZone) {
+    Objects.requireNonNull(text);
+    for (DateTimeFormatter f : dtFormatters) {
+      try {
+        return TimestampTZUtil.parse(text, defaultTimeZone, f);
+      } catch (DateTimeException e) {
```

Comment:
Try finally won't better in this case?

Reply:
Also, from pattern perspective a TryParse would be more elegant in this case. Of course it is just preferences, but I like this pattern. Way more descriptive from code reading perspective.
https://learn.microsoft.com/en-us/dotnet/api/system.int32.tryparse?view=net-8.0#system-int32-tryparse(system-string-system-int32@)

I know ref and out keyword are not exists in Java but with return type it is possible to handle. (Just FYI, no need to change)

Reply:
Thanks for the comments! Can you elaborate a bit on what you had in mind around try finally?

Reply:
```
  try {
        return TimestampTZUtil.parse(text, defaultTimeZone, f);
      } catch (DateTimeException e) {
      }
```

If the catch does not anything it is a bit cleaner if it is just try finally. (Again it is just my style :)  It is not a big deal. ) Just when I see a catch it would mean I will handle somehow an exception. If I do not do I would rather use finally.

Reply:
The `catch` is used to capture the `DateTimeException` and restrain/ignore it. Without the `catch` the exception will propagate which is not what we want here; the intention is to try all available parsers till we find one that can parse the value or till we run out of options.

Reply:
Oh I See. Thanks for the clarification.